serow_ros/src/serow_driver.cpp

```diff
@@ -91,85 +89,67 @@ class SerowDriver {
     }
 
     void joint_state_topic_callback(const sensor_msgs::JointState::ConstPtr& msg) {
-        joint_state_data_.push(*msg);
-        if (joint_state_data_.size() > 100) {
-            joint_state_data_.pop();
-            ROS_WARN("SEROW is dropping joint state measurements, SEROW estimate is not real-time");
-        }
+        joint_state_data_ = *msg;
+        gotJointMsg = true;
     }
 
     void base_imu_topic_callback(const sensor_msgs::Imu::ConstPtr& msg) {
-        base_imu_data_.push(*msg);
-        if (base_imu_data_.size() > 100) {
-            base_imu_data_.pop();
-            ROS_WARN("SEROW is dropping base IMU measurements, SEROW estimate is not real-time");
-        }
+        base_imu_data_ = *msg;
+        gotImuMsg = true;
     }
 
     void force_torque_state_topic_callback(const geometry_msgs::WrenchStamped::ConstPtr& msg) {
-        const std::string& frame_id = msg->header.frame_id;
-        ft_data_[frame_id].push(*msg);
-        
-        if (ft_data_.at(frame_id).size() > 100) {
-            ft_data_.at(frame_id).pop();
-            ROS_WARN("SEROW is dropping leg F/T measurements, SEROW estimate is not real-time");
-        }
+        ft_data_[msg->header.frame_id] = *msg;
     }
 
 private:
     void run() {
         ros::Rate loop_rate(loop_rate_); // Define the loop rate
         while (ros::ok()) { // Check if ROS is still running
-            if (joint_state_data_.size() > 0 && base_imu_data_.size() > 0) {
+            if (gotJointMsg && gotImuMsg) {
                 // Create the joint measurements
-                const auto& joint_state_data = joint_state_data_.front();
                 std::map<std::string, serow::JointMeasurement> joint_measurements;
-                for (size_t i = 0; i < joint_state_data.name.size(); i++) {
+                for (size_t i = 0; i < joint_state_data_.name.size(); i++) {
                     serow::JointMeasurement joint{};
                     joint.timestamp =
-                        static_cast<double>(joint_state_data.header.stamp.sec) +
-                        static_cast<double>(joint_state_data.header.stamp.nsec) * 1e-9;
-                    joint.position = joint_state_data.position[i];
-                    joint_measurements[joint_state_data.name[i]] = std::move(joint);
+                        static_cast<double>(joint_state_data_.header.stamp.sec) +
+                        static_cast<double>(joint_state_data_.header.stamp.nsec) * 1e-9;
+                    joint.position = joint_state_data_.position[i];
+                    joint_measurements[joint_state_data_.name[i]] = std::move(joint);
```

Comment:
we can also check if the `joint_state_data_` has the velocity specified to feed that directly to SEROW instead of having SEROW estimating it.